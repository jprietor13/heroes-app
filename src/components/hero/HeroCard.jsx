import React from 'react'
import { NavLink } from 'react-router-dom'

export const HeroCard = ({ id,
superhero, 
publisher, 
alter_ego,
first_appearance,
characters }) => {

  const heroImage = `/assets/images/${id}.jpg`

  return (
    <div className='col'>
      <div className='card'>
        <div className='row no-gutters'>
          <div className='col-4'>
            <img src={heroImage} alt={superhero} className="card-img-top" />
          </div>
          <div className='col-8'>
            <h4 className='card-title mt-2'>{superhero}</h4>
            <p className='card-text'>{alter_ego}</p>
            {alter_ego !== characters &&
              <p className='text-muted'>{characters}</p>
            }
            <p className='card-text'>{first_appearance}</p>
            <NavLink to={`${id}`}>
              Mas...
            </NavLink>
          </div>
        </div>
      </div>
    </div>
  )
}
