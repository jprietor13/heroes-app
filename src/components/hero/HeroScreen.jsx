import React, { useMemo } from 'react'
import { useParams, Navigate, useNavigate } from 'react-router-dom'
import { getHeroesById } from '../../helpers/getHeroesById'

export const HeroScreen = () => {

  const { idHero } = useParams();
  const navigate = useNavigate()

  const hero = useMemo(() => {
    return getHeroesById(idHero)
  }, [idHero]) 

  if(!hero){
    return <Navigate to="/" />
  }

  const imageHero = `/assets/images/${hero.id}.jpg`

  const handleReturn = () => {
    // if(hero.publisher === "DC Comics"){
    //   navigate('/dc', { replace: true })
    // } else {
    //   navigate('/marvel', { replace: true })
    // }
    navigate(-1)
  }

  return (
    <div className="row mt-5">
      <div className="col-4">
        <img src={imageHero} alt={hero.superhero} className="img-thumbnail"/>
      </div>
      <div className="col-8">
        <h3>{hero.superhero}</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item">
            <b>Alter ego: </b> {hero.alter_ego}
          </li>
          <li className="list-group-item">
            <b>Publisher: </b> {hero.publisher}
          </li>
          <li className="list-group-item">
            <b>First Appearance: </b> {hero.first_appearance}
          </li>
        </ul>
        <h5 className="mt-3">Characters</h5>
        <p>{hero.characters}</p>
        <button type="button" className="btn btn-primary" onClick={handleReturn}>
          Regresar
        </button>
      </div>
    </div>
  )
}
