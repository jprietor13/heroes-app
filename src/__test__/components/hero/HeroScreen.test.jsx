import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { HeroScreen } from "../../../components/hero/HeroScreen";

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate, 
}))

describe("Pruebas en <HeroScreen />", () => {

  const wrapper = mount(
    <MemoryRouter initialEntries={ ['/hero'] }>
      <Routes>
        <Route path="/hero" element={ <HeroScreen /> } />
        <Route path="/" element={ <h1>Hero Not Found</h1> } />
      </Routes>
    </MemoryRouter>
  )

  it("Debe cargarse corretamente", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('No debe de mostrar el HeroScreen si no hay un heroe en la URL', () => {
    expect(wrapper.find('h1').text().trim()).toBe('Hero Not Found')
  })

  it('Debe de mostrar un heroe si el parametros existe y se encuentra', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={ ['/hero/marvel-spider'] }>
        <Routes>
          <Route path="/hero/:idHero" element={ <HeroScreen /> } />
          <Route path="/hero" element={ <h1>Hero Not Found test</h1> } />
        </Routes>
      </MemoryRouter>
    )
 
    expect(wrapper.find('h3').exists()).toBe(true)
  })

})