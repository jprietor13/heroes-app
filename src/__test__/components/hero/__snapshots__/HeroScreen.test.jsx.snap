// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroScreen /> Debe cargarse corretamente 1`] = `
<h1>
  Hero Not Found
</h1>
`;
